pymc_marketing/mlflow.py

```diff
@@ -393,6 +396,69 @@ def log_sample_diagnostics(
     mlflow.log_param("arviz_version", posterior.attrs["arviz_version"])
 
 
+def log_loocv_metrics(
+    model: Model | MMM | None,
+    idata: az.InferenceData,
+) -> None:
+    """Log Bayesian LOOCV metrics to MLflow.
+
+    Compute Pareto-smoothed importance sampling leave-one-out cross-validation (PSIS-LOO-CV).
+    Estimates the expected log pointwise predictive density (elpd) using Pareto-smoothed
+    importance sampling leave-one-out cross-validation (PSIS-LOO-CV).
+    In order to compute the PSIS-LOO-CV, we need to compute the log_likelihood of the model too,
+    which will extend the inference data with a log_likelihood group.
+
+    Parameters
+    ----------
+    idata : az.InferenceData
+        The InferenceData object returned by the sampling method.
+
+    Sets
+    ----
+    idata.log_likelihood : az.InferenceData
+        The InferenceData object with the log_likelihood group, unless it already exists.
+    """
+    if "posterior" not in idata or "sample_stats" not in idata:
+        logging.warning(
+            "Skipping LOOCV metrics: InferenceData object does not contain\
+                         the group posterior and sample stats."
+        )
+        return
+
+    # try-excepts necessary in case of empty models, like `test_log_data_no_data`
+    if "log_likelihood" not in idata:
```

Comment:
I would suggest the computation is done separately from the logging. The if block makes it hard to extend for new models.

Reply:
Agree! Let's make functions do one thing only ;)